File: frontend/src/types/transportGuide.ts
export interface TransportGuide {
  id: string;
  cnpj: string;
  empresa: string;
  numeroColete?: string;
  numeroArma: string;
  calibre: string;
  qtdMunicoes: number;
  origem: string;
  destino: string;
  trajeto: string;
  motivo: string;
  arquivoGuia?: string;
  status: TransportGuideStatus;
  createdAt: string;
  updatedAt: string;
  createdBy: string;
  approvedBy?: string;
  approvedAt?: string;
}

export interface CreateTransportGuideDTO {
  cnpj: string;
  empresa: string;
  numeroColete?: string;
  numeroArma: string;
  calibre: string;
  qtdMunicoes: number;
  origem: string;
  destino: string;
  trajeto: string;
  motivo: string;
  arquivoGuia?: File;
}

export interface UpdateTransportGuideDTO extends Partial<CreateTransportGuideDTO> {
  id: string;
}

export interface TransportGuideFilters {
  cnpj?: string;
  empresa?: string;
  numeroArma?: string;
  calibre?: string;
  motivo?: string;
  status?: TransportGuideStatus;
  startDate?: string;
  endDate?: string;
  createdBy?: string;
}

export enum TransportGuideStatus {
  DRAFT = 'DRAFT',
  SUBMITTED = 'SUBMITTED',
  APPROVED = 'APPROVED',
  REJECTED = 'REJECTED',
  IN_TRANSIT = 'IN_TRANSIT',
  COMPLETED = 'COMPLETED',
  CANCELLED = 'CANCELLED'
}

export const TransportGuideStatusLabels = {
  [TransportGuideStatus.DRAFT]: 'Rascunho',
  [TransportGuideStatus.SUBMITTED]: 'Enviado',
  [TransportGuideStatus.APPROVED]: 'Aprovado',
  [TransportGuideStatus.REJECTED]: 'Rejeitado',
  [TransportGuideStatus.IN_TRANSIT]: 'Em Trânsito',
  [TransportGuideStatus.COMPLETED]: 'Concluído',
  [TransportGuideStatus.CANCELLED]: 'Cancelado'
};

export const TransportGuideStatusColors = {
  [TransportGuideStatus.DRAFT]: 'bg-gray-100 text-gray-800',
  [TransportGuideStatus.SUBMITTED]: 'bg-blue-100 text-blue-800',
  [TransportGuideStatus.APPROVED]: 'bg-green-100 text-green-800',
  [TransportGuideStatus.REJECTED]: 'bg-red-100 text-red-800',
  [TransportGuideStatus.IN_TRANSIT]: 'bg-yellow-100 text-yellow-800',
  [TransportGuideStatus.COMPLETED]: 'bg-green-100 text-green-800',
  [TransportGuideStatus.CANCELLED]: 'bg-gray-100 text-gray-800'
};

File: backend/src/main/java/com/z7design/fleet_manager/model/TransportGuide.java
package com.z7design.fleet_manager.model;

import com.z7design.fleet_manager.model.enums.TransportGuideStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "transport_guides")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class TransportGuide {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 18)
    private String cnpj;

    @Column(nullable = false)
    private String empresa;

    @Column(length = 50)
    private String numeroColete;

    @Column(nullable = false, length = 50)
    private String numeroArma;

    @Column(nullable = false, length = 20)
    private String calibre;

    @Column(nullable = false)
    private Integer qtdMunicoes;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String origem;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String destino;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String trajeto;

    @Column(nullable = false)
    private String motivo;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private TransportGuideStatus status;

    @Column(length = 500)
    private String arquivoGuiaPath;

    @Column(nullable = false)
    private String createdBy;

    private String approvedBy;

    private LocalDateTime approvedAt;

    private String rejectedBy;

    private LocalDateTime rejectedAt;

    private String rejectionReason;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
        if (status == null) {
            status = TransportGuideStatus.DRAFT;
        }
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}




























